src/main/java/com/yodle/vantage/component/dao/VersionDao.java

```diff
@@ -62,70 +45,38 @@ public boolean createNewVersion(String componentName, String version) {
         ).get("matched")).equals(false);
     }
 
-    private void createUnknownVersion(String component) {
-	createShadowVersion(component, UNKNOWN_VERSION);
-    }
-
-    private void createLatestVersion(String component) {
-	createShadowVersion(component, LATEST_VERSION);
-    }
-
-    private void createShadowVersion(String component, String type) {
+    public void createShadowVersion(String component, String type) {
         jdbcTemplate.update(
                 "MATCH (c:Component {name: {1}}) " +
-                        "MERGE (v:Version {version: {2}, unknown: true})-[:VERSION_OF]->(c) ON CREATE SET v.created=timestamp() ",
+                        "MERGE (v:Version {version: {2}, unknown: true})-[:VERSION_OF]->(c) " +
+                        "ON CREATE SET v.created=timestamp() " +
+                        //technically, we dont care about this functionally, but the concurrency of version creation requires
+                        //that we hold a write lock on all version nodes we'll be adding dependency relationships to or from
+                        //before we try adding any relationships so we can ensure we've taken the locks in a deterministic
+                        //deadlock-free order
+                        "ON MATCH SET v.matched=true ",
                 component, type
         );
     }
 
-    public boolean createResolvedImplicitDependency(Version version, Version dep, Collection<String> profiles) {
```

Comment:
creating resolved/requested dependencies now just involves creating the relationship link since we've moved verifying the version is already created to an earlier step of ComponentService